extensions/sockets/src/main/java/org/mule/module/socket/api/connection/AbstractSocketConnection.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.api.connection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Contains common configuration parameters to all the kinds of {@link SocketConnection}
+ *
+ * @since 4.0
+ */
+public abstract class AbstractSocketConnection implements SocketConnection
+{
+
+    protected final Logger LOGGER = LoggerFactory.getLogger(getClass());
+    protected final ConnectionSettings connectionSettings;
+    protected boolean wasDisconnected = false;
+
+    protected AbstractSocketConnection(ConnectionSettings connectionSettings)
+    {
+        this.connectionSettings = connectionSettings;
+    }
+
+    @Override
+    public void disconnect()
+    {
+        wasDisconnected = true;
+        if (LOGGER.isDebugEnabled())
```

Comment:
no point in this since the message is static anyway. This only makes sense when the message needs to be built or has parameters